### PR TITLE
Add Brake and Spinback effect to Hercules DJ Control MP3 e2 controller mapping

### PR DESCRIPTION
**Pull request by sblaisot**
This PR adds Brake and Spinback effect to Hercules DJ Control MP3 e2 mapping with supershift+backward and Supershift+Forward

### CODE REVIEW

File: res/controllers/Hercules DJ Control MP3 e2-scripts.js [hunk 1]
```diff
@@ -830,17 +830,30 @@ HerculesMP3e2.wind = function(midino, control, value, status, group) {
...
 
-    if (control == 0x21 || control == 0x0D) {
-        if (superButtonHold >= 1 && value) {
-            newValue = HerculesMP3e2.knobIncrement(deck, "pregain", 0, 4, 1, 20, 1);
-            engine.setValue(deck, "pregain", newValue);
+    if (control == 0x21 || control == 0x0D) { //forward
+
+        if (superButtonHold == 2) {
+            if (debug)
+                print("**** BRAKE deck " + (control == 0x0D ? deckA.toString() : deckB.toString()) + " : " + (value ? "true" : "false"));
```
Comment (ywwg) [inline res/controllers/Hercules DJ Control MP3 e2-scripts.js]: comment out debug lines, and use {} even on single-line if statements

Comment (ywwg) [inline res/controllers/Hercules DJ Control MP3 e2-scripts.js]: herp derp "if debug" -- ok no need to comment out, but still add {}

Comment (sblaisot) [inline res/controllers/Hercules DJ Control MP3 e2-scripts.js]: ok, done

**Comment by sblaisot** [comment]
wiki updated:
http://www.mixxx.org/wiki/doku.php/hercules_dj_control_mp3_e2

**Comment by ywwg** [comment]
Is it our practice now to run the js linter over these things just to make sure, or should I just merge it in?  @Be-ing ?

**Comment by sblaisot** [comment]
Nevermind, I already used JSHint and JSBeautifier on it ;)

**Comment by illuusio** [comment]
How have this worked before these new Spinback if stuff?

**Comment by sblaisot** [comment]
@illuusio what do you mean ? the mapping worked well before and works well after this change.

**Comment by illuusio** [comment]
Yes I assumed so but this mostly adds some ifs with new  MIDI commands I'm wondering why they are there.

**Comment by sblaisot** [comment]
let me try to explain a bit more.

This mapping has 3 layers:
- The normal layer, when you just press a button
- The shift layer, when you stay pressed on the automix button used as a shift key then press a button
- The supershift layer when you stay pressed on the automix + the scratch button

Before this change, the backward/forward button had the same effect with shift and supershift (superButtonHold >= 1).
This change add brake and spinback effect on supershift-backward and supershift-forward (superButtonHold == 2).
normal and shifted backward and forward don't change (superButtonHold == 1 and else cases)

This change also corrects a race condition on shift-backward release changing 
if (superButtonHold >= 1 && value) {
with
if (superButtonHold == 1) {
    if (value) {

to avoid going through "else" when unneeded.

Hope this is clear now.

**Comment by sblaisot** [comment]
to which line are you referring when you say "new MIDI commands"
There is absolutely no new "MIDI command" in this change.

If you are referring to lines 934-944 (of the new file), there is no new command, only new brackets as asked by @ywwg to have brackets on on-liner ifs

**Comment by illuusio** [comment]
@sblaisot  Sorry my bad. Yes they were just corrected lines. Ok now I understand more correctly what this does.

**Comment by sblaisot** [comment]
Ready to merge...

**Comment by illuusio** [comment]
LGTM so merge

**Comment by sblaisot** [comment]
Thanks !

**Pull request merged**